src/ccutil/ccutil.cpp

```diff
@@ -33,68 +31,74 @@ CCUtil::CCUtil()
 CCUtil::~CCUtil() = default;
 
 /**
- * @brief CCUtil::main_setup - set location of tessdata and name of image
+ * @brief Finds the path to the tessdata directory.
  *
- * @param argv0 - paths to the directory with language files and config files.
- * An actual value of argv0 is used if not nullptr, otherwise TESSDATA_PREFIX is
- * used if not nullptr, next try to use compiled in -DTESSDATA_PREFIX. If
- * previous is not successful - use current directory.
- * @param basename - name of image
+ * This function determines the location of the tessdata directory based on the
+ * following order of precedence:
+ * 1. If `argv0` is provided, use it.
+ * 2. If `TESSDATA_PREFIX` environment variable is set and the path exists, use
+ *    it.
+ * 3. On Windows, check for a "tessdata" directory in the executable's directory
+ *    and use it.
+ * 4. If `TESSDATA_PREFIX` is defined at compile time, use it.
+ * 5. Otherwise, use the current working directory.
+ *
+ * @param argv0 argument to be considered as the data directory path.
+ * @return The path to the tessdata directory or current directory.
  */
-void CCUtil::main_setup(const std::string &argv0, const std::string &basename) {
-  imagebasename = basename; /**< name of image */
-
-  const char *tessdata_prefix = getenv("TESSDATA_PREFIX");
-
-  // Ignore TESSDATA_PREFIX if there is no matching filesystem entry.
-  if (tessdata_prefix != nullptr && !std::filesystem::exists(tessdata_prefix)) {
-    tprintf("Warning: TESSDATA_PREFIX %s does not exist, ignore it\n", tessdata_prefix);
-    tessdata_prefix = nullptr;
-  }
-
+std::filesystem::path find_data_path(const std::string &argv0) {
+  // If argv0 is set, always use it even if it is not a valid directory
   if (!argv0.empty()) {
-    /* Use tessdata prefix from the command line. */
-    datadir = argv0;
-  } else if (tessdata_prefix) {
-    /* Use tessdata prefix from the environment. */
-    datadir = tessdata_prefix;
-#if defined(_WIN32)
-  } else if (datadir.empty() || !std::filesystem::exists(datadir)) {
-    /* Look for tessdata in directory of executable. */
-    char path[_MAX_PATH];
-    DWORD length = GetModuleFileName(nullptr, path, sizeof(path));
-    if (length > 0 && length < sizeof(path)) {
-      char *separator = std::strrchr(path, '\\');
-      if (separator != nullptr) {
-        *separator = '\0';
-        std::string subdir = path;
-        subdir += "/tessdata";
-        if (std::filesystem::exists(subdir)) {
-          datadir = subdir;
-        }
-      }
+    std::filesystem::path path(argv0);
+    if (!std::filesystem::exists(path) ||
```

Comment:
It should be sufficient to test the 2nd part of the condition (`!std::filesystem::is_directory(path)`).